### ISSUE TITLE
I have a question? Why not transfer label to one hot. 

### ISSUE THREAD

**Issue by TianlinZhang668** [jungwhank/rcnn-text-classification-pytorch#1]
Why not transfer label to one hot. 
The original paper has softmax.

and if we use cross_entropy
we need to transfer label to one hot. 

Thx a lot

**Comment by jungwhank** [comment]
Hi, @18222278799 

I think your question is related to PyTorch's `crossentropy` loss.
https://github.com/jungwhank/rcnn-text-classification-pytorch/blob/aed84f66514322745c4ae004fc3cff772d6f1c0c/trainer.py#L20

This is PyTorch's official document of `crossentropy` loss.
https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss

As you can see in document, PyTorch's `CrossEntropyLoss` combines `LogSoftmax()` and `NLLLoss()`. So we don't need softmax layer because it will be computed in our loss function. Also, we don't need to transfer label to one-hot because PyTorch's `CrossEntropyLoss` expects integer labels (check `target` shape in document).

Thx !
